dll/win32/shell32/CDefView.cpp

```diff
@@ -1866,6 +1866,9 @@ SelectExtOnRename(void)
     HKEY hKey;
     LONG error;
     DWORD dwValue = FALSE, cbValue;
+    static DWORD s_dwCached = 0xDEADBEEF;
```

Comment:
Shouldn't we use a global to be initialized at startup, as it may be done for other such settings ?

Reply:
Wouldn't the caching mechanism require an explorer-reboot to ever reread that setting? Isn't that over-optimization?
I base my argumentation on the assumption that SelectExtOnRename() is only called only once per renaming-request.

Reply:
In principle, registry caching should be done together with listening for registry changes (registry notifications) on the involved registry keys. However this mechanism is currently unimplemented in ReactOS and thus cannot be used.

Reply:
Typically settings like this would be in something like `SHELLSTATE`, however this looks like a custom ROS thing, so not sure how to handle that.

Reply:
Well then a vote? I'd vote for leaving the caching out for now. We could always touch it later, when Vista+ stuff designs will get more clear to us. For now just go with the most simple code. After all we also had no performance issue before PR3993. So what to fear?

Reply:
Now deleted the caching.

Reply:
@JoachimHenze death by a thousand cuts?
There are tons of unoptimized code paths in our shell, sure.
When drawing each entry, we check the disk, we check the registry, all slowing us down a tiny bit.
Don't start complaining when explorer is slow on refreshing icons or browsing folders then.

Reply:
Mark, as said, I expected that function to only be called when invoking a rename. Is that a wrong assumption?

Reply:
This particular function is indeed only called on rename, but `SHELL_FS_HideExtension` for example is not.

Reply:
Should we modify `SHELL_FS_HideExtension`?